### DISCUSSION TITLE
Suggest for component generator

### DISCUSSION THREAD

**Discussion by CeSun** [Dreamescaper/BlazorBindings.Maui#97] [General]
It is recommended that component generators support custom component namespaces or rename components.

Because some components may have the same name as MAUI components, such as Button and Label, but because components such as ContentPage must be used in general, the namespace of MAUI components is used by default, and there will be ambiguous references at this time.
I tried: @using MD = xxxx.xxx.Button; <MD.Button></MD.Button> but couldn't find the correct component.

So I hope that the component generator supports adding custom prefixes/suffixes to components or supports renaming.

**Comment by Dreamescaper** [comment]
I hoped that this wouldn't be needed, because it is actually a razor issue, that it doesn't understand aliases.
https://github.com/dotnet/razor/issues/7670

Unfortunately, seems like it won't be fixed anytime soon, which makes this a perfectly reasonable request.

**Reply by CeSun** [reply]
So.... do you take this suggest? 😀

**Reply by Dreamescaper** [reply]
Ability to rename components in generator is not perfect, it leads to bunch of other potential issues (like finding the proper base type), so I hoped to avoid that. If I won't be able to find any other option in the near future - I will implement that.

**Comment by Dreamescaper** [comment]
I have merged the support for Aliases property to rename the type name itself. It will be included in the next release, but you can try it with the [nightly release](https://dreamescaper.github.io/MobileBlazorBindingsDocs/maui-blazor-bindings/contribute/nightly-builds.html).

You can take a look [here](https://github.com/Dreamescaper/BlazorBindings.Maui/blob/main/samples/ThirdPartyControlsSample/Properties/Elements.cs) for an example.
